### PR TITLE
doc: Project Build information

### LINKED ISSUE #89: Project Build information in README
Currently, the project [README](https://github.com/hyperledger/aries-framework-go/blob/master/README.md) file doesn't contain build instructions. Need to add basic project build instructions to the README file.

### PR DESCRIPTION
**Pull request by rolsonquadras**
Document project build info/targets in README along with prerequisites

Closes #89 

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,13 +10,25 @@
 ## Introduction
 A go framework for Aries
 
-## Contributing
+## Build
+##### Prerequisites
+- Go 1.12
+- Docker v18.09.2
```
Comment (troyronda) [inline README.md]: This seems very specific?

Comment (rolsonquadras) [reply README.md]: @troyronda I'll update to '>v18.09.2'.

File: README.md [hunk 2]
```diff
@@ -10,13 +10,25 @@
 ## Introduction
 A go framework for Aries
```
Comment (troyronda) [inline README.md]: While you are here, I think "go" should be "Go".

File: README.md (hunk 1 above)
Comment (rolsonquadras) [reply README.md]: @troyronda on a second thought, there is not really a version requirement for Docker. Hence [removed](https://github.com/hyperledger/aries-framework-go/compare/9e512eec533a9845abfe2604f7f636c61c9284a9..77de294918da55729525425ed777814198cf215f#diff-04c6e90faac2675aa89e2176d2eec7d8R16) the version details.

File: README.md (hunk 2 above)
Comment (rolsonquadras) [reply README.md]: @troyronda [Updated](https://github.com/hyperledger/aries-framework-go/compare/77de294918da55729525425ed777814198cf215f..480655f1ce24de110974569fbec272fa8c20411f)

File: README.md [hunk 3]
```diff
@@ -8,15 +8,27 @@
...
 
 ## Introduction
-A go framework for Aries
+A Go framework for Aries
 
-## Contributing
+## Build
+##### Prerequisites
+- Go 1.12
+- Docker
```
Comment (troyronda) [inline README.md]: Docker and Make are for running tests. Note: openssl is also used to generate test artifacts.

File: README.md [hunk 4]
```diff
@@ -8,15 +8,27 @@
 # aries-framework-go
 
 ## Introduction
-A go framework for Aries
+A Go framework for Aries
 
-## Contributing
+## Build
+##### Prerequisites
+- Go 1.12
```
Comment (troyronda) [inline README.md]: Note: this is the version of Go our tests are currently running against.

File: README.md (hunk 3 above)
Comment (rolsonquadras) [reply README.md]: [Updated](https://github.com/hyperledger/aries-framework-go/pull/91/files#diff-04c6e90faac2675aa89e2176d2eec7d8R14) the subheading.

File: README.md (hunk 4 above)
Comment (rolsonquadras) [reply README.md]: [Added](https://github.com/hyperledger/aries-framework-go/pull/91/files#diff-04c6e90faac2675aa89e2176d2eec7d8R14) 'test' word in the subheading.

File: README.md [hunk 5]
```diff
@@ -8,15 +8,28 @@
 # aries-framework-go
 
 ## Introduction
-A go framework for Aries
+A Go framework for Aries
 
-## Contributing
+## Build
+##### Prerequisites for test
```
Comment (troyronda) [inline README.md]: Any particular reason for 5th level sub-heading? (#####)

perhaps "for running tests"?

File: README.md [hunk 6]
```diff
@@ -8,15 +8,28 @@
...
+- Go 1.12
```
Comment (troyronda) [inline README.md]: Of course, also Go 1.12 is also a general requirement :).

Comment (rolsonquadras) [reply README.md]: [Updated](https://github.com/hyperledger/aries-framework-go/compare/4114ed97adddd4bae7ca7aa37395dd5030ab4c59..bc1bec9caa24a8ee97ec09d6762fe66a2e5bdefb#diff-04c6e90faac2675aa89e2176d2eec7d8R14).

File: README.md (hunk 5 above)
Comment (rolsonquadras) [reply README.md]: [Updated](https://github.com/hyperledger/aries-framework-go/compare/4114ed97adddd4bae7ca7aa37395dd5030ab4c59..bc1bec9caa24a8ee97ec09d6762fe66a2e5bdefb#diff-04c6e90faac2675aa89e2176d2eec7d8R17).

There is no specific reason as such(i was considering ideas between having it bolded or as subheading and decided with 5th level heading) :) let me know, i can change this.

File: README.md [hunk 7]
```diff
@@ -8,15 +8,31 @@
...
 
-Found a bug? Ready to submit a PR? Want to submit a proposal for your grand
-idea? Follow our [guidelines](.github/CONTRIBUTING.md) for more information
-to get you started!
+##### Prerequisites (for running tests)
+- Go 1.12
+- Docker
+- Make
+- OpenSSL
```
Comment (llorllale) [inline README.md]: We're using docker image `frapsoft/openssl` by default. The developer may override this by setting the `OPENSSL_CMD` enviroment variable.

Comment (rolsonquadras) [reply README.md]: [Removed](https://github.com/hyperledger/aries-framework-go/compare/bc1bec9caa24a8ee97ec09d6762fe66a2e5bdefb..1c758ac9c8a5372da72239e742bd1fd802b7b515).

File: README.md [hunk 8]
```diff
@@ -8,15 +8,30 @@
...
+
+# run linter checks
+make checks
+
+# run unit tests
+make unit-test
+```
 
-### Crypto Material generation for tests
+##### Crypto Material generation for tests
```
Comment (troyronda) [inline README.md]: inconsistent capitalization (Crypto **M**aterial)

Comment (rolsonquadras) [reply README.md]: didn't notice this as it was an existing line. [Updated](https://github.com/hyperledger/aries-framework-go/compare/1c758ac9c8a5372da72239e742bd1fd802b7b515..45fafadc2e71b354f413c11155c1c1f6f9212c03#diff-04c6e90faac2675aa89e2176d2eec7d8R34) now.

**Pull request merged**